### DISCUSSION TITLE
using my video to run speed

### DISCUSSION THREAD

**Discussion by althonmp** [roboflow/supervision#833] [Q&A]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

HI ,i used my video to run speed [speed_estimation](https://github.com/roboflow/supervision/tree/develop/examples/speed_estimation) code.
I didnt do anything with code,and my video had a little bit proble.could you help me ?  
 issue >>
 AttributeError: 'NoneType' object has no attribute 'reshape'

my video            [https://www.youtube.com/watch?v=8Gvz_FjWy4s](url)
my video run result  [https://youtu.be/0KxiJQKj-vA?si=lVrhGR3edo499JP5](https://youtu.be/0KxiJQKj-vA?si=lVrhGR3edo499JP5)

![1706612267809](https://github.com/roboflow/supervision/assets/122033865/9f9a37d3-67b0-4b82-a8a8-b2b8077ddb22)
`    def transform_points(self, points: np.ndarray) -> np.ndarray:
        reshaped_points = points.reshape(-1, 1, 2).astype(np.float32)
        transformed_points = cv2.perspectiveTransform(reshaped_points, self.m)
        return transformed_points.reshape(-1, 2)
`

**Comment by SkalskiP** [comment]
Hi @althonmp 👋🏻 ! Could you tell me where you found this version of the code? I'm asking because some other users faced this issue already, and `transform_points` got updated.

```
def transform_points(self, points: np.ndarray) -> np.ndarray:
    if points.size == 0:
        return points

    reshaped_points = points.reshape(-1, 1, 2).astype(np.float32)
    transformed_points = cv2.perspectiveTransform(reshaped_points, self.m)
    return transformed_points.reshape(-1, 2)
```

**Reply by althonmp** [reply]
this issue already, and transform_points got updated.  i got updated. i finished it

**Comment by althonmp** [comment]
> Hi @althonmp 👋🏻 ! Could you tell me where you found this version of the code? I'm asking because some other users faced this issue already, and `transform_points` got updated.
> 
> ```
> def transform_points(self, points: np.ndarray) -> np.ndarray:
>     if points.size == 0:
>         return points
> 
>     reshaped_points = points.reshape(-1, 1, 2).astype(np.float32)
>     transformed_points = cv2.perspectiveTransform(reshaped_points, self.m)
>     return transformed_points.reshape(-1, 2)
> ```
HI@SkalskiP
the code found from https://github.com/roboflow/supervision/tree/develop/examples/speed_estimation
I will try upgrade code,

this is my video details
![1706775111703](https://github.com/roboflow/supervision/assets/122033865/0f47d0d7-a5b1-46fc-9890-7caaa66e3734)

**Comment by SkalskiP** [comment]
It looks to me that you need to pull the latest version of the code on the `develop` branch.

**Reply by althonmp** [reply]
OK

**Reply by althonmp** [reply]
Hi 👋, Piotr
I pull the latest version of the code on the develop branch.  the code run test ok
![1707999437750](https://github.com/roboflow/supervision/assets/122033865/21834965-3e3e-4014-9f87-68ed9e98c5e2)

but speed estimation is not good .
my new vehicles result215https://www.youtube.com/watch?v=ZlSYMHs9Hpo

May i  try to adjust the SOURCE and TARGET configuration?

**Comment by SkalskiP** [comment]
Let me convert this issue into a discussion and move it to the Q&A section.

**Reply by althonmp** [reply]
OK

**Comment by althonmp** [comment]
@SkalskiP Thank you for your response.
i try to adjust the SOURCE and TARGET configuration.the road is completely flat.The  speed estimation hadn't a little bit good.my  [result215](https://www.youtube.com/watch?v=ZlSYMHs9Hpo)


I showed that the area surrounded by the source region of interest measures approximately 10 meters wide and 200 meters long.
[223](https://www.youtube.com/watch?v=iTJNYLB2_9k)

How do i know to  camera angle, altitude, or viewing point ?
